cloudos_cli/__main__.py

```diff
@@ -2175,6 +2179,429 @@ def run_bash_job(ctx,
+        project_name = array_file_project
+
+    # this needs to be in another call to datasets, by default it uses the global project name
+    if custom_script_project is None:
+        custom_script_project = project_name
+
+    # setup separators for API and array file (the're different)
+    separators = {
+        ",": { "api": ",", "file": "," },
+        ";": { "api": "%3B", "file": ";" },
+        "space": { "api": "+", "file": " " },
+        "tab": { "api": "tab", "file": "tab" },
+        "|": { "api": "%7C", "file": "|" }
+    }
+
+    # Setup datasets
+    try:
+        ds = Datasets(
+            cloudos_url=cloudos_url,
+            apikey=apikey,
+            workspace_id=workspace_id,
+            project_name=project_name,
+            verify=verify_ssl,
+            cromwell_token=None
+        )
+        if custom_script_project is not None:
+            # If a custom script project is specified, create a new Datasets object for it
+            # This allows the user to run custom scripts in a different project
+            ds_custom = Datasets(
+                cloudos_url=cloudos_url,
+                apikey=apikey,
+                workspace_id=workspace_id,
+                project_name=custom_script_project,
+                verify=verify_ssl,
+                cromwell_token=None
+            )
+    except BadRequestException as e:
+        if 'Forbidden' in str(e):
+            print('[Error] It seems your call is not authorised. Please check if ' +
+                                'your workspace is restricted by Airlock and if your API key is valid.')
+            sys.exit(1)
+        else:
+            raise e
+
+    # Split the array_file path to get the directory and file name
+    p = Path(array_file)
+    directory = str(p.parent)
+    file_name = p.name
+
+    # fetch the content of the directory
+    result = ds.list_folder_content(directory)
+
+    # retrieve the S3 bucket name and object key for the specified file
+    for file in result['files']:
+        if file.get("name") == file_name:
+            array_file_id = file.get("_id")
+            s3_bucket_name = file.get("s3BucketName")
+            s3_object_key = file.get("s3ObjectKey")
+            s3_object_key_b64 = base64.b64encode(s3_object_key.encode()).decode()
+            break
+    else:
+        raise ValueError(f'File "{file_name}" not found in the "Data" folder of the project "{project_name}".')
+
+    # retrieve the metadata of the array file
+    headers = {
+        "Content-type": "application/json",
+        "apikey": apikey
+    }
+    url = (
+        f"{cloudos_url}/api/v1/jobs/array-file/metadata"
+        f"?separator={separators[separator]['api']}"
+        f"&s3BucketName={s3_bucket_name}"
+        f"&s3ObjectKey={s3_object_key_b64}"
+        f"&teamId={workspace_id}"
+    )
+    r = retry_requests_get(url, headers=headers, verify=verify_ssl)
+    if r.status_code >= 400:
+        raise BadRequestException(r)
```

Comment:
Could this be a new method for Job class? A rule of thumb for __main.nf__ is: if it contains an API call, wrap it in one or more methods of a given class

Reply:
Added as a new method in Job class